walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/TimePoint.java

```diff
@@ -0,0 +1,73 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps;
+
+import java.util.Comparator;
+
+/**
+ * Class representing a point of time (either a start of an event or an end of it).
+ */
+public final class TimePoint {
```

Comment:
optional: I don't think this class is really needed. You could achieve something similar by using the TimeRange and merging and splitting ranges. But since you have created this, there is no need to remove it.

Reply:
I created it to store number of optional attendees (for the optional task) but mainly to separate events start and end I wasn't sure if it is possible to achieve with TimeRange.

Reply:
Yes, I initially didn't notice the field `numberOfOptional`, which you would indeed need to store somewhere.